### PR TITLE
fix for KeyError: 'epoch'

### LINKED ISSUE #41: Unable to start basic training with pretrained coco weights, KeyError: 'epoch'.
Hey, thanks for the repo. 
I wanted to train coco on 1280 and set my python train file like this:

```
from ultralytics import YOLOv10

model = YOLOv10(f'yolov10x.pt')

results = model.train(data='coco.yaml', epochs=120, imgsz=1280, device=[0, 1, 2, 3], 
                      batch=16, close_mosaic=20, project='coco-1280', resume=False)

```
this should start standard training of coco with 1280 imgsz from pretrained weight file of 640 imgsz.
However, when I try to load model like this I get:
```

Traceback (most recent call last):                                                                                                                                                                                  
  File "/home/.config/Ultralytics/DDP/_temp_asyw3_vx140544015144176.py", line 12, in <module>                                                                                                                 
    results = trainer.train()                                                                                                                                                                                       
  File "/home/anaconda3/envs/yolov10/lib/python3.9/site-packages/ultralytics/engine/trainer.py", line 213, in train                                                                                           
    self._do_train(world_size)                                                                                                                                                                                      
  File "/home/anaconda3/envs/yolov10/lib/python3.9/site-packages/ultralytics/engine/trainer.py", line 327, in _do_train                                                                                       
    self._setup_train(world_size)                                                                                                                                                                                   
  File "/home/anaconda3/envs/yolov10/lib/python3.9/site-packages/ultralytics/engine/trainer.py", line 319, in _setup_train                                                                                    
    self.resume_training(ckpt)                                                                                                                                                                                      
  File "/home/anaconda3/envs/yolov10/lib/python3.9/site-packages/ultralytics/engine/trainer.py", line 665, in resume_training                                                                                 
    start_epoch = ckpt["epoch"] + 1                                                                                                                                                                                 
KeyError: 'epoch'  
```

for some reason it is trying to resume even when it should not.

### PR DESCRIPTION
**Pull request by tfriedel**
fixes #41

### CODE REVIEW

**Comment by jameslahm** [comment]
Thanks for your contribution!

File: ultralytics/engine/trainer.py [hunk 1]
```diff
@@ -225,9 +225,9 @@ def _setup_ddp(self, world_size):
         torch.cuda.set_device(RANK)
         self.device = torch.device("cuda", RANK)
         # LOGGER.info(f'DDP info: RANK {RANK}, WORLD_SIZE {world_size}, DEVICE {self.device}')
-        os.environ["NCCL_BLOCKING_WAIT"] = "1"  # set to enforce timeout
+        os.environ["TORCH_NCCL_BLOCKING_WAIT"] = "1"  # set to enforce timeout
         dist.init_process_group(
-            "nccl" if dist.is_nccl_available() else "gloo",
+            backend="nccl" if dist.is_nccl_available() else "gloo",
```
Comment (jameslahm) [inline ultralytics/engine/trainer.py:230]: Could you please revert these changes that are less relevant to this issue?

Comment (tfriedel) [reply ultralytics/engine/trainer.py]: ok. those changes were in the newer ultralytics repo and I saw some warnings and at first wasn't sure if they also had something to do with the issue. I guess not.
